staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -768,6 +770,120 @@ func (r *Request) Watch(ctx context.Context) (watch.Interface, error) {
 	}
 }
 
+type WatchListResult struct {
+	// err holds any errors we might have received
+	// during streaming.
+	err error
+
+	// store holds collected data
```

Comment:
expand here about how the streaminglist explaning why the watch for this list can actually get update and delete watch.Events.